tests/providers/amazon/aws/triggers/test_serialization.py

```diff
@@ -0,0 +1,372 @@
+            ),
+            RedshiftDeleteClusterTrigger(
+                cluster_identifier=TEST_CLUSTER_IDENTIFIER,
+                poll_interval=WAITER_DELAY,
+                max_attempts=MAX_ATTEMPTS,
+                aws_conn_id=AWS_CONN_ID,
+            ),
+            RdsDbAvailableTrigger(
+                db_identifier=TEST_DB_INSTANCE_IDENTIFIER,
+                waiter_delay=WAITER_DELAY,
+                waiter_max_attempts=MAX_ATTEMPTS,
+                aws_conn_id=AWS_CONN_ID,
+                region_name=AWS_REGION,
+                response=TEST_RESPONSE,
+                db_type=RdsDbType.INSTANCE,
+            ),
+            RdsDbDeletedTrigger(
+                db_identifier=TEST_DB_INSTANCE_IDENTIFIER,
+                waiter_delay=WAITER_DELAY,
+                waiter_max_attempts=MAX_ATTEMPTS,
+                aws_conn_id=AWS_CONN_ID,
+                region_name=AWS_REGION,
+                response=TEST_RESPONSE,
+                db_type=RdsDbType.INSTANCE,
+            ),
+            RdsDbStoppedTrigger(
+                db_identifier=TEST_DB_INSTANCE_IDENTIFIER,
+                waiter_delay=WAITER_DELAY,
+                waiter_max_attempts=MAX_ATTEMPTS,
+                aws_conn_id=AWS_CONN_ID,
+                region_name=AWS_REGION,
+                response=TEST_RESPONSE,
+                db_type=RdsDbType.INSTANCE,
+            ),
+            SqsSensorTrigger(
+                sqs_queue=TEST_SQS_QUEUE,
+                aws_conn_id=AWS_CONN_ID,
+                max_messages=TEST_MAX_MESSAGES,
+                num_batches=TEST_NUM_BATCHES,
+                wait_time_seconds=TEST_WAIT_TIME_SECONDS,
+                visibility_timeout=TEST_VISIBILITY_TIMEOUT,
+                message_filtering=TEST_MESSAGE_FILTERING,
+                message_filtering_match_values=TEST_MESSAGE_FILTERING_MATCH_VALUES,
+                message_filtering_config=TEST_MESSAGE_FILTERING_CONFIG,
+                delete_message_on_reception=TEST_DELETE_MESSAGE_ON_RECEPTION,
+                waiter_delay=WAITER_DELAY,
+            ),
+            StepFunctionsExecutionCompleteTrigger(
+                execution_arn=TEST_ARN,
+                aws_conn_id=AWS_CONN_ID,
+                waiter_delay=WAITER_DELAY,
+                waiter_max_attempts=MAX_ATTEMPTS,
+                region_name=AWS_REGION,
+            ),
+        ],
+        ids=gen_test_name,
+    )
+    def test_serialize_recreate(self, trigger):
+        # generate the DB object from the trigger
+        trigger_db: Trigger = Trigger.from_object(trigger)
+
+        # use the same json serializer as the one that is used for the DB
+        jsonifier = ExtendedJSON()
+        # apply the function that processes the data pre-insert
+        json_params = jsonifier.process_bind_param(trigger_db.kwargs, Mock())
+        # apply the function that processes the data post-select
+        retrieved_params = jsonifier.process_result_value(json_params, Mock())
```

Comment:
I think we could use here `BaseSerialization.serialize` and  `BaseSerialization.deserialize` rather then SQLAlchemy's TypeDefinition which implementation depends on DB Type

Reply:
hmm, yes. If `ExtendedJSON` changes then the test would be desynchronized from reality, but it would also make the code less twisted...

Reply:
I guess when ExtendedJSON serious changed and would not use BaseSerialisation then we have Airflow 3 🥳 🥳 

My main point here then better provide actual Dialect from session, e.g. we could get it by `session.bind.dialect` then it should use use serialisation depend on DB type and we safe here.

Reply:
yeah I kinda wanted to do that at first, but as soon as I involved an actual `session` it started to pull so much code I needed to mock that I didn't manage to do it in a clean way.

Reply:
Ok let's do that later, it's just tests and we know that it just call serialize/deserialize and some hacks for MsSQL which would soon removed